### ISSUE TITLE
A11y Review - Alert - Expandable variant loses heading a11y role

### ISSUE THREAD

**Issue by TimRoe** [department-of-veterans-affairs/vets-design-system-documentation#5365]
## Configuring this issue
- [ ] Add issue to appropriate epic
## Description
While probing changes as part of #5345 it was noted that for the expandable Alert the heading role was being overruled by the tab role and expanded state a11y state. As far as I can tell, this is expected behavior since a single focusable element can't be both a "tab" role and "heading" role simultaneously. This behavior was signed off on by the prior a11y specialist for the mobile team.

That said, it does seem potentially problematic navigationally to be unable to jump to the header with an Alert if it's expandable while you can if it's any non-expandable variant (including dismissable since the dismiss action is a separate target).

For visual users it makes sense for this functionality to be combined and touching anywhere on the entire collapsed Alert will expand it (and anywhere on the "header" portion to collapse when expanded). For screen reader, these potentially should be separate focus elements--the header text and then separately the expand/collapse piece. This does, however, have a cost in additional focus targets going from 1 to 2.

## Tasks
- [ ] Review behavior in mobile
- [ ] Compare with expandable Alert a11y behavior in web 
- [ ] Determine if the current behavior is adequate or should be improved

## Acceptance Criteria
- [ ] Either comment on this ticket determining we're satisfied with the existing behavior for now or create a follow-up ticket to update the Alert component to better meet desired a11y behavior
